include/zephyr/drivers/dma.h

```diff
@@ -557,6 +562,11 @@ static inline int z_impl_dma_request_channel(const struct device *dev,
 	/* dma_context shall be the first one in dev data */
 	struct dma_context *dma_ctx = (struct dma_context *)dev->data;
 
+	/* Driver has overridden the default implementation */
+	if (api->request_channel) {
```

Comment:
is it possible that move to application to handle the soc level? as for DMA driver, it does not know how many request need to shared/can be shared. it is related to SOC integartion.